### ISSUE TITLE
Always seems to use yarn

### ISSUE THREAD

**Issue by johnwc** [madskristensen/NpmTaskRunner#91]
**Describe the bug**
Just recently I noticed the icon in task runner explorer is a blue silhouette of a cat. Instead of red npm icon. Every time I run a task it runs it with yarn. We do not use yarn on this project and have no yarn file in the same folder as package.json, I can't figure out why it is using yarn instead of npm. I even add `"packageManager": "npm@9.0.0",` to the package.json.

**To Reproduce**
I'm really not sure how to reproduce, as I am unaware of why it is thinking it should use yarn.

**Expected behavior**
The tasks run using npm.

**Screenshots**
<img width="164" alt="image" src="https://github.com/madskristensen/NpmTaskRunner/assets/2798441/3e71e48b-dcc4-4de1-aab6-7877ed8946d4">

**Comment by dionisisZyg** [comment]
Hello @johnwc 
Did you  by any chance got any more information on the problem?

Thanks you.

**Comment by johnwc** [comment]
@dionisisZyg no, even on a new laptop with fresh VS install, it still uses yarn

**Comment by dionisisZyg** [comment]
Hmm..
The documentation and the packagemanager seem to care for .yarn files in order to pick app YARN as the package manager.
This is very weird. I am under the impression that this stopped working after a certain VS version update because some time ago it was working fine.
My version is: 17.9.3 Community 2022 (64-bit) 

Thanks @johnwc  for your reply.
Did you resort to other solutions for your development environment?
Personally for the time being i am executing a poweshell script in order to save time..

**Comment by johnwc** [comment]
It installs and runs using yarn. It is just that we did not install yarn, we prefer the output and options of npm as that is what we are used to. Are you a member of this repo?

**Comment by dionisisZyg** [comment]
Hello again, @johnwc .

No, i am not a member of this repo.
I am using the extension in order to speed up the development process.

Today it seems to pick up NPM as the package manager correctly!
What i did:

- Uninstalled and reinstalled the extension
- Update VS to the latest version
- Uninstalled and reinstalled globally yarn
- restarted pc

![image](https://github.com/madskristensen/NpmTaskRunner/assets/29570517/cb16f1d1-d8df-406a-b48b-7907d578d29b)

**Comment by johnwc** [comment]
We never installed yarn.

**Comment by johnwc** [comment]
@dionisisZyg @madskristensen I finally got fed up with not knowing why it was using yarn and tracked down the issue. The version of the extension that is published to the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.NPMTaskRunner) is an older release that is based on last commit of 35e9949. Within that version of the code base, the check for the three files `.yarnclean`, `.yarnrc`, or `yarn.lock` does a reverse hierarchy search starting from the directory that contains `package.json` back to the root drive. If it finds any of those three files in the paths, it uses yarn. In my case, I had `.yarnrc` in my profile directory.

https://github.com/madskristensen/NpmTaskRunner/blob/35e99495b7f8ed27d6f5c630fa762fbf406faf45/src/TaskRunner/TaskRunnerProvider.cs#L169-L182

There seems to be a newer released version of the extension available from the [Open VSIX Gallery](https://www.vsixgallery.com/) site. It seems be the latest codebase, and this issue does not exists in it.
https://www.vsixgallery.com/extension/c5203f5f-4785-4c09-b383-688a100a9f5d

**Comment by johnwc** [comment]
@dionisisZyg  @madskristensen actually this still exists in the latest release. Should it be doing that, looking for yarn files outside of project's working directory? If someone uses both yarn and npm or has these files in their path and not realize it, they will not be able to use npm and no option to manually set it to only use npm.
